src/blight/actions/bitcode_extract.py

```diff
@@ -0,0 +1,51 @@
+"""
+The `BitcodeExtract` action.
+"""
+
+import hashlib
+import logging
+import os
+import subprocess
+from pathlib import Path
+from typing import List
+
+from blight.action import CompilerAction
+from blight.enums import Lang
+from blight.tool import CompilerTool
+
+logger = logging.getLogger(__name__)
+
+
+class BitcodeExtract(CompilerAction):
+    """
+    Compile and extract bitcode. The output bitcode file will be located placed in directory
+    specified by `store=/some/dir/`. The setting is passed in the `FindActions` configuration.
+    If `store` is not specified the action will error.
+    """
+
+    def before_run(self, tool: CompilerTool) -> None:  # type: ignore
+        store = self._config.get("store")
+
+        if store is None:
+            logger.debug("not extracting bitcode to an unspecified location")  # pragma: no cover
+            assert False  # pragma: no cover
+
+        if tool.lang not in [Lang.C, Lang.Cxx]:
+            logger.debug("not extracting bitcode for an unknown language")
```

Comment:
```suggestion
            logger.debug("not extracting bitcode for an unknown language")
            return
```

Reply:
Alternatively, it might make sense to drop this check and check against `tool.stage` instead, `return`ing if we're not on a `CompileObject` stage.